packages/components/src/DataTable/Header/DataTableHeaderCell.tsx

```diff
@@ -53,11 +54,12 @@ const DataTableHeaderCellLayout = forwardRef(
   ) => {
     const { onSort } = useContext(DataTableContext)
 
-    const handleClick = () => {
+    const onClick = () => {
       if (onSort && canSort) {
         onSort(id, sortDirection === 'asc' ? 'desc' : 'asc')
       }
     }
+    const clickableProps = useClickable({ onClick })
```

Comment:
To @jhardy 's point about `role="button"`, you can exclude it like so:
```suggestion
    const {role, ...clickableProps} = useClickable({ onClick })
```